ci/Dockerfile

```diff
@@ -1,142 +1,37 @@
-        cd Python-${PYTHON35VER}/ && ./configure && make && make install ; \
-    fi
+ARG FULLDEPS_TESTING=0
+ENV FULLDEPS_TESTING=${FULLDEPS_TESTING}
 
-RUN if [ "${FULLDEPS_TESTING}" = "1" ]; then \
-        export PYTHON36VER="3.6.9" ; \
-        wget --no-verbose https://www.python.org/ftp/python/${PYTHON36VER}/Python-${PYTHON36VER}.tgz ; \
-        tar zxf Python-${PYTHON36VER}.tgz ; \
-        cd Python-${PYTHON36VER}/ && ./configure && make && make install ; \
-    fi
+# install other python versions for tox testing
 
 RUN if [ "${FULLDEPS_TESTING}" = "1" ]; then \
-        export PYTHON38VER="3.8.0" ; \
-        export PYTHONSUBVER="b3"; \
-        wget --no-verbose https://www.python.org/ftp/python/${PYTHON38VER}/Python-${PYTHON38VER}${PYTHONSUBVER}.tgz ; \
-        tar zxf Python-${PYTHON38VER}${PYTHONSUBVER}.tgz ; \
-        cd Python-${PYTHON38VER}${PYTHONSUBVER}/ && ./configure && make && make install ; \
-    fi
-
-# remove symlinks to newly installed pythons
-RUN cd /usr/local/bin; \
-    rm -f 2to3; \
-    rm -f python3; \
-    rm -f python3-config; \
-    rm -f pydoc3; \
-    rm -f pip3; \
-    rm -f pyvenv;
-
-# install dependencies from toolchain source build
-
-RUN if [ "${TOOLCHAIN_FLAVOR}" = "src" ]; then \
-        apt-get install -y autoconf autogen bison dejagnu \
-                           flex flip gawk git gperf gzip nsis \
-                           openssh-client p7zip-full perl python-dev \
-                           libisl-dev tcl tofrodos zip \
-                           texinfo texlive texlive-extra-utils; \
-    fi
-
-# download toolchain
-
-ENV TOOLCHAIN_LONGVER=gcc-arm-none-eabi-9-2019-q4-major
-ENV TOOLCHAIN_SUBDIR="9-2019q4/RC2.1"
-ENV TOOLCHAIN_URL=https://developer.arm.com/-/media/Files/downloads/gnu-rm/${TOOLCHAIN_SUBDIR}/${TOOLCHAIN_LONGVER}-${TOOLCHAIN_FLAVOR}.tar.bz2
-ENV TOOLCHAIN_HASH_linux=bcd840f839d5bf49279638e9f67890b2ef3a7c9c7a9b25271e83ec4ff41d177a
-ENV TOOLCHAIN_HASH_src=f162a655f222319f75862d7aba9ff8a4a86f752392e4f3c5d9ef2ee8bc13be58
-
-# extract toolchain
-
-RUN cd /opt && wget --no-verbose ${TOOLCHAIN_URL}
-RUN cd /opt && echo "${TOOLCHAIN_HASH_linux} ${TOOLCHAIN_LONGVER}-x86_64-linux.tar.bz2\n${TOOLCHAIN_HASH_src} ${TOOLCHAIN_LONGVER}-src.tar.bz2" | sha256sum -c --ignore-missing
-RUN cd /opt && tar xfj ${TOOLCHAIN_LONGVER}-${TOOLCHAIN_FLAVOR}.tar.bz2
-
-# build toolchain (if required)
-
-RUN if [ "${TOOLCHAIN_FLAVOR}" = "src" ]; then \
-        pushd /opt/${TOOLCHAIN_LONGVER} ; \
-        ./install-sources.sh --skip_steps=mingw32 ; \
-        ./build-prerequisites.sh --skip_steps=mingw32 ; \
-        ./build-toolchain.sh --skip_steps=mingw32,manual ; \
-        popd ; \
+        nix-env -iP python3-3.8.2 ; \
+        nix-env --set-flag priority 6 python3-3.8.2 ; \
+        nix-env -iP python3-3.6.10 ; \
+        nix-env --set-flag priority 7 python3-3.6.10 ; \
+        nix-env -iP python3-3.5.9 ; \
+        nix-env --set-flag priority 8 python3-3.5.9 ; \
```

Comment:
Not sure if this is the best way how to get Python 3.{5,6,8} into the cointainer. (Python 3.7 is already installed, because it is a dependency of pipenv).

Reply:
If we do not need it in the current profile, an alternative could be just cache it into the store using 
 ` nix-shell -p python36 --run echo`, etc. But your approach works fine too.

Reply:
Python tests need to find the `python3.x` binaries in `$PATH`